selfconference/selfconf-android
# 2019 Dependencies/Libraries Update

## What does this PR do❓
This PR updates the dependencies and libraries of the self.conference Android app to be compatible with 2019 tooling. In particular, the following are library upgrades that were recommended by the library developers:
- Support Library -> AndroidX
- RxJava 1 -> RxJava 2
- Latest Square open source updates (OkHttp, Retrofit, Moshi, Picasso)
- Dagger 1 -> Dagger 2
- Robolectric 2.4 -> Robolectric 4.2.1

Before the changes made in this PR, I could not build the project in the latest Android Studio. The first commit (19e4085) in this PR fixes that. This PR addresses Android Studio/build tools incompatibility, deprecation warnings, and a few other minor warnings. The app now has minimum SDK 21 (Lollipop) to take advantage of the newest tools and dependencies. Very likely that self.conference attendees with Android phones will be on Lollipop or later.

Finally, this PR updates the conference dates to those of self.conference 2019, allowing users to actually see the sessions for June 7 and June 8!

## How to test it 🔬
1. Clone the code in the [2019-update](https://github.com/selfconference/selfconf-android/tree/2019-update) branch.
1. Build the code with the latest version of Android Studio (3.4.1 for macOS as of this writing). You should face no build errors and only a few warnings.
1. Run the app on an emulator or an Android device (Lollipop and higher). You should see sessions, speakers, and sponsors for the 2019 conference, and be able to "submit feedback". The app is behaviorally the same as previous years (hence version 2.0.10), so expect it to function the same!

## Any background info you would like to include ✅
Hi! 👋 This is my first PR to the selfconf-android project. My name's Matt, an Android/Java developer at Detroit Labs. Macklin and Amber set up this project really nicely, and I just wanted to get it into ship shape for 2019. 🚢

## Screenshots (if applicable) 📷
![selfconf-android-2019-preview](https://user-images.githubusercontent.com/8700989/58279016-4db21580-7d6b-11e9-8383-c0be2ef04e4f.gif)

## Acknowledgements
Special thanks to @BritneyS for the PR template!
